cohortextractor/query_engines/mssql.py

```diff
@@ -287,14 +296,27 @@ def apply_filter(self, query, filter_node):
         # Does this filter require another table? i.e. is the filter value itself an
         # Output node, which has a source that we may need to include here
         value_expr, other_table = self.get_value_expression(filter_node.value)
+
         if other_table is not None:
-            query = self.include_joined_table(query, other_table)
+            if isinstance(value_expr, sqlalchemy.Column) and operator_name in [
+                "in_",
+                "not_in",
+            ]:
+                # For an in_/not_in query that uses a query from another table, convert
+                # the Column to a select query, and select over only the rows for each patient
+                value_expr = sqlalchemy.select(value_expr)
+                value_expr = value_expr.where(
+                    query.selected_columns.patient_id == other_table.c.patient_id
+                ).correlate(other_table)
```

Comment:
So I think this this explicitly selects just the matching rows on patient id for the in/not-in.  The tests work without the `correlate` bit, so I'm not sure if it's necessary.  It adds the base table column selection into the subquery, but since it's the base table, maybe it's never needed, and just the where clause is OK?  I may just be being misled by the existence of a `correlate` sqlalchemy method